packages/bot/src/scratch/blocks/Binary/Trade Definition/trade_definition.js

```diff
@@ -83,35 +85,47 @@ Blockly.Blocks.trade_definition = {
         };
     },
     onchange(event) {
-        //        if (!ScratchStore.instance.root_store.core.ui.is_dark_mode_on) {
         // TODO: incomment this when the dark mode is done
+        // if (!ScratchStore.instance.root_store.core.ui.is_dark_mode_on) {
         setBlockTextColor(this);
-        //        }
+        // }
 
+        // We don't add "this.workspace.isDragging()" condition here as we want to immediately remove
+        // other instances of this block when dragged out of the workspace.
         if (!this.workspace || this.isInFlyout) {
             return;
         }
 
         if (event.type === Blockly.Events.BLOCK_CREATE && event.ids.includes(this.id)) {
             // Maintain single instance of this block, dispose of older ones.
-            const top_blocks = this.workspace.getTopBlocks(false);
-
-            top_blocks.forEach(top_block => {
+            this.workspace.getTopBlocks().forEach(top_block => {
                 if (top_block.type === this.type && top_block.id !== this.id) {
-                    top_block.dispose(false);
+                    runIrreversibleEvents(() => {
```

Comment:
I think this should be reversible. In case user want to undo back to the old `trade_definition` block ?

Reply:
That will its own event, in this case we want to ensure the user doesn't undo the delete event.